test/samplers/ratelimiting_sampler_test.js

```diff
@@ -59,10 +59,35 @@ describe ('RateLimitingSampler should', () => {
         let sampler = new RateLimitingSampler(0.1);
         sampler._rateLimiter._balance = 1;
 
-        assert.equal(sampler.isSampled('operation', {}), true, 'expected decision to be true');
+        assert.isTrue(sampler.isSampled('operation', {}), 'expected decision to be true');
 
         clock = sinon.useFakeTimers(initialDate + 10000);
-        assert.equal(sampler.isSampled('operation', {}), true, 'expected decision to be true');
+        assert.isTrue(sampler.isSampled('operation', {}), 'expected decision to be true');
         clock.restore();
     });
+
+    it ('should update successfully', () => {
+        let initialDate = new Date(2011,9,1).getTime();
+        let clock = sinon.useFakeTimers(initialDate);
+        let sampler = new RateLimitingSampler(1.0);
+        sampler._rateLimiter._balance = 1;
```

Comment:
Abstraction leak - you could add initBalance optional parameter to the sampler and avoid having to access private members.